### ISSUE TITLE
Credentials not escaping command arguments.

### ISSUE THREAD

**Issue by mterzo** [jenkinsci/puppet-jenkins#526]
If any of the arguments passed on the command line to the groovy script has a apostrophe in the content the command will fail. 

I ran into this with a password with an apostrophe. I think issue may extend to other functionality.

**Comment by jhoblitt** [comment]
The new `jenkins_user` type [also] handles this correctly. https://github.com/jenkinsci/puppet-jenkins/blob/master/NATIVE_TYPES_AND_PROVIDERS.md#jenkins_user

**Comment by mterzo** [comment]
Oh, that's way better.

**Comment by jhoblitt** [comment]
It would be great if you could give it a try and see if it works for you.  This is a legit bug and we do need to either fix `jenkins::user` or turn it into a wrapper around the native type.

**Comment by mterzo** [comment]
I'm using the jenkins_credentials type.

I'm running into a load error for the puppet_x requires.  I'm running puppet 3.8.x  on my agent and hosts.  There's a workaround I've seen used in another plugin, that plugin references # JJM WORK_AROUND #14073 and #7788

I'm essentially having to do this:

````
begin
  require 'puppet_x/jenkins/type/cli'
rescue LoadError => detail
  require 'pathname' # JJM WORK_AROUND #14073 and #7788
  require Pathname.new(__FILE__).dirname + "../../" + "puppet_x/jenkins/type/cli"
end
````

I'm not fully tested yet, once I do i'll submit a PR.

**Comment by jhoblitt** [comment]
This is the upstream bug https://tickets.puppetlabs.com/browse/SERVER-973, I believe I tested the workaround with both puppetserver 1.x and 2.x.  I don't use puppetserver in production at this point so it is possible the situation has changed.

**Comment by mterzo** [comment]
Yeah, i read that.  Not very helpful.  I got the all the imports happy, however i'm running into

Error: Could not find a suitable provider for jenkins_credentials

**Comment by jhoblitt** [comment]
That likely means the cli jar isn't present or your missing the `retries` gem.  You could try manually installing the `retries` gem in the agent's env.

**Comment by mterzo** [comment]
Was the retires gem.

**Comment by mterzo** [comment]
All good now.

**Comment by jhoblitt** [comment]
I'd like to leave this open since this issue is still present in `jenkins::user`

**Comment by elconas** [comment]
We are running in to the same problem. I can confirm the workarount "# JJM WORK_AROUND #14073 and #7788" works. I can also confirm that I needed to install the "retries" gem. 

Questions: 

- Why is the retries gem not installed by Jenkins Module as part of enabeling the provider ?
- How is this incompatibility between jenkins module and puppetserver addressed ? (Jenkins module implementing the workaround ? Moving your code to lib ? )

Regards
